compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -928,9 +928,7 @@ impl<'hir> intravisit::Map<'hir> for Map<'hir> {
     }
 }
 
-pub(super) fn index_hir<'tcx>(tcx: TyCtxt<'tcx>, cnum: CrateNum) -> &'tcx IndexedHir<'tcx> {
-    assert_eq!(cnum, LOCAL_CRATE);
-
+pub(super) fn index_hir<'tcx>(tcx: TyCtxt<'tcx>, (): ()) -> &'tcx IndexedHir<'tcx> {
```

Comment:
Why have a `(): ()` parameter instead of simply removing it?

Reply:
The query system is designed to work with queries taking exactly two arguments: the TyCtxt and a key. I am not convinced that removing the extra `()` parameter is worth the added complexity in the query system. (I am not even sure this is possible without a sizeable modification to the query plumbing.)